tools/setup_helpers/ninja_builder.py

```diff
@@ -61,10 +62,57 @@ def patch(obj, attr_name, val):
             finally:
                 setattr(obj, attr_name, orig_val)
 
-        orig_compile = distutils.unixccompiler.UnixCCompiler._compile
-        orig_link = distutils.unixccompiler.UnixCCompiler.link
+        if self.compiler.compiler_type == 'msvc':
+            import distutils.msvccompiler
+            import distutils.msvc9compiler
+            if sys.version[0] == 2:
+                orig_compiler = distutils.msvc9compiler.MSVCCompiler
+            else:
+                orig_compiler = distutils._msvccompiler.MSVCCompiler
+            orig_compile = orig_compiler.compile
+            orig_link = orig_compiler.link
+            orig_spawn = orig_compiler.spawn
+        else:
+            orig_compiler = distutils.unixccompiler.UnixCCompiler
+            orig_compile = orig_compiler._compile
+            orig_link = orig_compiler.link
+
+        def win_compile(self, sources,
+                        output_dir=None, macros=None, include_dirs=None, debug=0,
+                        extra_preargs=None, extra_postargs=None, depends=None):
 
-        def _compile(self, obj, src, ext, cc_args, extra_postargs, pp_opts):
+            def spawn(cmd):
+                # Using regex to match src and obj
+
+                src_regex = re.compile('/T(p|c)(.*)')
+                src_list = [m.group(2) for m in (
+                    src_regex.match(elem) for elem in cmd) if m]
+
+                obj_regex = re.compile('/Fo(.*)')
+                obj_list = [m.group(1) for m in (
+                    obj_regex.match(elem) for elem in cmd) if m]
+
+                if len(src_list) >= 1 and len(obj_list) >= 1:
+                    src = src_list[0]
+                    obj = obj_list[0]
+                else:
+                    # Cannot find src or obj, revert back to original style
+                    orig_spawn(cmd)
+                    return
+
+                builder.writer.build(
+                    [obj], 'compile', [src],
+                    variables={
+                        'cmd': cmd,
+                        'deps': 'msvc'
```

Comment:
This patch looks good to me now, but I'm not sure how dep tracking works on windows. What does setting it to `msvc` mean? The whole point of using ninja was to get fast builds with correct dependency graph

Reply:
@apaszke [This](https://ninja-build.org/manual.html#_deps) is the discussion about it in the manual of Ninja build. The flag `deps=msvc` here has nothing to do with dependencies, it just makes the build process faster according to the description in the manual.